src/hera_display_games/core/sprites.py

```diff
@@ -0,0 +1,57 @@
+from abc import ABC, abstractmethod
+from . import map_dict
+
+DIR_DICT = {
+    "ul": [0, 1],
+    "ur": [1, 1],
+    "r": [1, 0],
+    "dr": [0, -1],
+    "dl": [-1, -1],
+    "l": [-1, 0],
+}
+
+
+class Sprite(ABC):
+    """Class for sprites."""
+
+    def __init__(self, location=[0, 0], color=[0, 255, 0], id=None):
+        """Init for sprites."""
+        self.location = location
+        self.color = color
+        self.dead = False
+        self.id = id
+
+    def move(self, movement):
+        try:
+            self.location[0] += DIR_DICT[movement][0]
+            self.location[1] += DIR_DICT[movement][1]
+        except TypeError:
+            if type(movement) == int:
+                self.location[0] = map_dict.reverse_led_map[movement][0]
+                self.location[1] = map_dict.reverse_led_map[movement][1]
+            else:
+                if len(movement) != 2:
+                    raise ValueError("movement should be a 2-tuple!")
+                self.location = movement.copy()
+        except KeyError:
+            raise ValueError("That was a bad string for movement")
```

Comment:
Maybe I am just not seeing it. I looked at the map_dict and there just seems to not be a lot of `"dead"` pixels. How does this handle a sprite attempting to move out of bounds?

Reply:
Hmmm, yeah I don't think that's being captured properly. Will fix.